docs/csharp/language-reference/compiler-messages/nullable-warnings.md

```diff
@@ -127,6 +127,10 @@ ms.date: 02/20/2025
 ---
 # Resolve nullable warnings
 
+The purpose of nullable warnings is to minimize the chance that your application throws a <xref:System.NullReferenceException?displayProperty=nameWithType> when run. To achieve this goal, the compiler uses static analysis and issues warnings when your code has constructs that might lead to null reference exceptions. You provide the compiler with information for its static analysis by applying type annotations and attributes. These annotations and attributes describe the nullability of arguments, parameters, and members of your types. In this article, you learn different techniques to address the nullable warnings the compiler generates from its static analysis. The techniques described here are for general C# code. Learn to work with nullable reference types and Entity Framework core in [Working with nullable reference types](/ef/core/miscellaneous/nullable-reference-types).
+
+Nullable reference types, including the operators `?` and `!` are allowed only when the nullable context is set to `enable` or `annotations`. You can set the nullable annotation using the `Nullable` [compiler option](../compiler-options/language.md#nullable) in your project file, or using the [`#nullable`](../preprocessor-directives.md#nullable-context) pragma in your source code.
```

Comment:
```suggestion
Nullable reference types, including the operators `?` and `!` are allowed only when the nullable context is set to `enable` or `annotations`. You can set the nullable context using the `Nullable` [compiler option](../compiler-options/language.md#nullable) in your project file, or using the [`#nullable`](../preprocessor-directives.md#nullable-context) pragma in your source code.
```

it looks like “context” was meant to be used rather than “annotations” here.

Reply:
Also, the language uses the term “nullable directive” rather than “nullable pragma”. Elsewhere in our docs we appear to call it a directive as well. There may be a reason that the term pragma is preferred here though. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#nullable-context